### PR TITLE
feat: adding docker scanning github action workflow

### PR DESCRIPTION
**Pull request by abiju-czi**

### CODE REVIEW

File: .github/actions/container-scanning/action.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    description: 'whether to fail the action if vulnerabilities are found'
+    required: false
+    default: "true"
+runs:
+  using: "composite"
+  steps:
+    - uses: aws-actions/configure-aws-credentials@v3
+      with:
+        aws-region: us-west-2
+        role-to-assume: arn:aws:iam::871040364337:role/ci/github-actions-inspector
```
Comment (hspitzley-czi) [inline .github/actions/container-scanning/action.yml:37]: Is this supposed to be the sec account or should it be core-platform-prod (where the images are pushed)

File: .github/actions/container-scanning/action.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        high_threshold:   ${{ inputs.high_threshold }}
+        medium_threshold:   ${{ inputs.medium_threshold }}
+        low_threshold:  ${{ inputs.low_threshold }}
+        other_threshold: ${{ inputs.other_threshold }}
+        display_vulnerability_findings: "enabled"
+    - name: Display Inspector vulnerability scan results (markdown)
+      shell: bash
+      run: cat ${{ steps.inspector.outputs.inspector_scan_results_markdown }}
+    - name: Fail job if vulnerability threshold is exceeded
+      if: "contains(inputs.fail_on_vulnerabilities, 'true')"
```
Comment (hspitzley-czi) [inline .github/actions/container-scanning/action.yml:56]: I don't think this needs the wrapping quotes
```suggestion
      if: contains(inputs.fail_on_vulnerabilities, 'true')
```

File: .github/actions/container-scanning/action.yml (hunk 1 above)
Comment (abiju-czi) [reply .github/actions/container-scanning/action.yml:37]: This is independent of accounts. It scans the image that's built in the pipeline. So I am using the sec account

File: .github/actions/container-scanning/action.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: Container Scanning
+description: 'A GitHub Action to scan a container image for security vulnerabilities that follows CZI Best Practices'
+inputs:
+  image-tag:
```
Comment (hspitzley-czi) [inline .github/actions/container-scanning/action.yml:4]: Nit: make consistent underscore vs hyphen in inputs

Comment (hspitzley-czi) [reply .github/actions/container-scanning/action.yml:4]: Also, is this the full repo + tag? If so maybe there's a more apt name

**Review by hspitzley-czi — APPROVED** [review]

**Pull request merged**